### ISSUE TITLE
False positives for exporting plugins to base classes of the same package

### ISSUE THREAD

**Issue by ipa-mdl** [fkie/catkin_lint#40]
`catkin_lint` reports an error if no dependences on the the plugins' base classes were listed, which is great.
However, this leads to false postives if the base class is part of the exporting packages and suggests the packages to depend on themselves.

Examples from  `ros_canopen`:
```
canopen_402: warning: package exports canopen_402 plugin but does not exec_depend on 'canopen_402'
     *         Your package exports a plugin for another package, but
     * fails to list said         package as a dependency.
     * You can ignore this problem with --ignore plugin_depend
canopen_master: warning: package exports canopen_master plugin but does not exec_depend on 'canopen_master'
socketcan_interface: warning: package exports socketcan_interface plugin but does not exec_depend on 'socketcan_interface' 
```

**Comment by ipa-mdl** [comment]
duplicate of #39, my bad :-/
Works with current master, sorry for the noise!

**Comment by roehling** [comment]
This reminds me that I probably should release a new version soon.
